src/installer/destinations.py

```diff
@@ -176,20 +208,32 @@ def write_script(
         with io.BytesIO(data) as stream:
             entry = self.write_to_fs(Scheme("scripts"), script_name, stream)
 
-            path = os.path.join(self.scheme_dict[Scheme("scripts")], script_name)
+            path = self._path_with_destdir(Scheme("scripts"), script_name)
             mode = os.stat(path).st_mode
             mode |= (mode & 0o444) >> 2
             os.chmod(path, mode)
 
             return entry
 
+    def _compile_bytecode(self, scheme: Scheme, record: RecordEntry) -> None:
+        """Compile bytecode for a single .py file."""
+        if scheme not in ("purelib", "platlib"):
+            return
+
+        target_path = self._path_with_destdir(scheme, record.path)
+        for level in self.bytecode_optimization_levels:
+            kwargs: Dict[str, Any] = {}
+            if sys.version_info >= (3, 9):  # pragma: no cover
+                kwargs["stripdir"] = str(self.destdir)
```

Comment:
Not sure if this causes any practical difference but this results in a path without the leading `/` being embedded, i.e. `usr/lib/python...` rather than the usual `/usr/lib/python...`.

Reply:
I think I've worked out a fix for this (using `ddir` instead of `stripdir`), which also means we can make the same call across all supported Python versions. Can you try the latest commit and see if the paths are correct now?

Reply:
It looks to be correct now. Thanks!